From 2cfbef438049fd4c3f73d1562773ad1f93900897 Mon Sep 17 00:00:00 2001
From: Dmitri Goutnik <dgoutnik@gmail.com>
Date: Fri, 20 May 2022 08:07:03 -0500
Subject: [PATCH] cmd/cgo: recognize clang 14 DWARF type names

Fixes #53013

Change-Id: I169d4eb2420a6da52cc9abe17da98c3092a91be6
Reviewed-on: https://go-review.googlesource.com/c/go/+/407514
Auto-Submit: Ian Lance Taylor <iant@google.com>
Reviewed-by: Cherry Mui <cherryyz@google.com>
Reviewed-by: Ian Lance Taylor <iant@google.com>
Run-TryBot: Ian Lance Taylor <iant@google.com>
TryBot-Result: Gopher Robot <gobot@golang.org>
---

diff --git a/src/cmd/cgo/gcc.go b/src/cmd/cgo/gcc.go
index 4d1a5bd..8ce5d4d 100644
--- a/src/cmd/cgo/gcc.go
+++ b/src/cmd/cgo/gcc.go
@@ -2242,6 +2242,8 @@
 	"long long unsigned int": "ulonglong",
 	"signed char":            "schar",
 	"unsigned char":          "uchar",
+	"unsigned long":          "ulong",     // Used by Clang 14; issue 53013.
+	"unsigned long long":     "ulonglong", // Used by Clang 14; issue 53013.
 }
 
 const signedDelta = 64
